spec/controllers/v1/legislators_controller_spec.rb

```diff
@@ -22,38 +19,60 @@
     end
 
     context "good email" do
-      context "new user" do
-        before { get :index, { email: 'user@example.com' } }
+      let!(:campaign) { FactoryGirl.create(:campaign_with_reps, count: 3, priority: 1) }
+      let (:state)    { FactoryGirl.create(:state, abbrev: 'CA') }
+      let (:district) { FactoryGirl.create(:district, district: '13', state: state) }
+      let!(:senator)  { FactoryGirl.create(:senator, state: state, with_us: false) }
+      let!(:rep)      { FactoryGirl.create(:representative, district: district,
+                                                            with_us: true) }
+      let(:campaign_ids) { campaign.legislators.pluck(:id) }
+      let(:returned_ids) { assigns(:target_legislators).map{|tl| tl['id']} }
```

Comment:
I don't think fixtures and factories should be used together. Factories are meant to take the place of fixtures. As you've written it above, I might have just used a before block and then `@lala` for each variable that needed to actually be called in your tests.  Still, I wouldn't worry too much. You might focus on using the before block rather than so many lets, but I don't think either makes a critical difference right now.

Generally, Rails controller specs make it easy to mix what should really be acceptance or integration tests in with what should be more functional testing.  In the ideal situation, we'd use mocks in functional controller testing - testing isolated bits of interaction without retesting what should be tested in unit tests in the model.  In our app, I wrote out several such mocks in the calls_controller spec if you want to see an example.

If we stick to skinny controllers, there are a lot fewer interactions to test in your controller and only a few objects that you actually need to mock out. Mocking takes a bit of time to learn to do well and quickly, but usually it saves you time over the long run and better isolates the behavior your trying to test.

Keep in mind, with `FactoryGirl.create` you're actually hitting the DB for each of these tests, slowing things down - especially as test suites grow. Sometimes, `FactoryGirl.build` can be used instead, but it won't persist data.  Ideally, factories would be used to build real database records only when absolutely needed.  By this nature, factory build objects will mostly be found in acceptance/feature or integration testing (although, I think I broke this rule at least once) and avoided elsewhere.

Make sense? I hope that's helpful.